packages/p2p/src/components/buy-sell/buy-sell-modal.jsx

```diff
@@ -99,7 +121,32 @@ const BuySellModal = ({ table_type, selected_ad, should_show_popup, setShouldSho
         />
     );
     const [is_account_balance_low, setIsAccountBalanceLow] = React.useState(false);
+    const [show_market_rate_change_error_modal, setShowMarketRateChangeErrorModal] = React.useState(false);
+    const [has_rate_changed_recently, setHasRateChangedRecently] = React.useState(false);
     const formik_ref = React.useRef();
+    const MAX_ALLOWED_RATE_CHANGED_WARNING_DELAY = 1000;
+
+    React.useEffect(() => {
+        const disposer = reaction(
+            () => buy_sell_store?.advert?.rate,
+            () => {
+                setHasRateChangedRecently(true);
+                setTimeout(() => {
+                    setHasRateChangedRecently(false);
+                }, MAX_ALLOWED_RATE_CHANGED_WARNING_DELAY);
```

Comment:
why do we need this?

Reply:
There are 2 scenarios in which the rate change is reflected to the modal:
1. User clicks `Confirm` button, rate change has not reflected yet by BE and the Mobx observable, API sends back error `OrderCreateFailRateChanged`, in this case I just check if there's this error then show the modal
2. Another case is when BE has reflected the rate change and Mobx also reflects it to the modal, user clicks `Confirm` button just milliseconds after rate changes (in this case they won't notice it since its very sudden), from QA subtask we also need to show the rate change error modal for this, so `MAX_ALLOWED_RATE_CHANGED_WARNING_DELAY` is a preset time value that is the period between the exact time where the rate change is reflected and when the user is able to submit their order by clicking `Confirm`. If the user clicks `Confirm` in this time period, we will show the rate change error modal

Reply:
For instance (for `MAX_ALLOWED_RATE_CHANGED_WARNING_DELAY = 1000`):
1. Seller changes their rate
2. BE and modal reflects the rate change
3. But buyer clicks the `Confirm` button `400ms` after the rate changes
4. In this case we will show the rate change error modal since 1000ms has not passed yet after rate change is reflected
Another case:
1. Seller changes rate
2. Rate change is reflected on BE and Mobx store
3. Buyer clicks `Confirm` 1050ms after rate changes
4. In this case we wont show the rate change error modal, since this time period should allow the buyer to notice this rate change in the modal